crates/ruff_server/src/server.rs

```diff
@@ -284,18 +285,24 @@ pub(crate) enum SupportedCodeAction {
 }
 
 impl SupportedCodeAction {
-    /// Returns the possible LSP code action kind(s) that map to this code action.
-    fn kinds(self) -> Vec<CodeActionKind> {
+    /// Returns the LSP code action kind that map to this code action.
+    fn to_kind(self) -> &'static CodeActionKind {
+        static QUICK_FIX: CodeActionKind = CodeActionKind::QUICKFIX;
+        static SOURCE_FIX_ALL: CodeActionKind = crate::SOURCE_FIX_ALL_RUFF;
+        static SOURCE_ORGANIZE_IMPORTS: CodeActionKind = crate::SOURCE_ORGANIZE_IMPORTS_RUFF;
         match self {
-            Self::QuickFix => vec![CodeActionKind::QUICKFIX],
-            Self::SourceFixAll => vec![CodeActionKind::SOURCE_FIX_ALL, crate::SOURCE_FIX_ALL_RUFF],
-            Self::SourceOrganizeImports => vec![
-                CodeActionKind::SOURCE_ORGANIZE_IMPORTS,
-                crate::SOURCE_ORGANIZE_IMPORTS_RUFF,
-            ],
+            Self::QuickFix => &QUICK_FIX,
+            Self::SourceFixAll => &SOURCE_FIX_ALL,
+            Self::SourceOrganizeImports => &SOURCE_ORGANIZE_IMPORTS,
         }
     }
 
```

Comment:
Nit: I think I would return the owned value here (or remove the `static` aliases)
```suggestion
    fn to_kind(self) -> CodeActionKind {
        match self {
            Self::QuickFix => CodeActionKind::QUICKFIX,
            Self::SourceFixAll => crate::SOURCE_FIX_ALL_RUFF,
            Self::SourceOrganizeImports => crate::SOURCE_ORGANIZE_IMPORTS_RUFF,
        }
    }

```